### ISSUE TITLE
using multiple databases

### ISSUE THREAD

**Issue by jamesdixon** [cilindrox/hapi-ioredis#8]
Hello again!

I recently added another redis database to my app that stores different data. Initially, I tried using the `select()` command to switch databases, but it appears to switch the database for the entire client meaning that all subsequent operations will reference the specified database. Given the global nature of the client created, this could cause major issues.

Have you dealt with this?

If not, I was thinking that maybe the plugin could be updated to be registered multiple times and named so that different clients could be accessed. For example:

```js
{
        plugin: {
            register: 'hapi-ioredis',
            options: {
                name: 'redisSessionAuthClient',
                redis: {
                    host: process.env.REDIS_HOST,
                    port: process.env.REDIS_PORT,
                    password: process.env.REDIS_PASSWORD,
                    database: 0,
                    enableReadyCheck: true,
                    tls: _.indexOf(['development', 'test'], process.env.NODE_ENV) !== -1 ? false : true
                }
            }
        }
    },

{
        plugin: {
            register: 'hapi-ioredis',
            options: {
                name: 'redisApiAuthClient',
                redis: {
                    host: process.env.REDIS_HOST,
                    port: process.env.REDIS_PORT,
                    password: process.env.REDIS_PASSWORD,
                    database: 1,
                    enableReadyCheck: true,
                    tls: _.indexOf(['development', 'test'], process.env.NODE_ENV) !== -1 ? false : true
                }
            }
        }
    }
```

These could then be accessed via:

```js
const { redisSessionAuthClient } = request;
const { redisApiAuthClient } = request;
```

Thoughts?

**Comment by cilindrox** [comment]
Hi there, and apologies for the delay. 

Haven't run into this use case myself, but definitely sounds like a good feature to add.

There's already [a `multiple` setting][1] that allows you to register the plugin several times on a given Hapi instance. From the looks of it, not setting the actual db when dialing the connection sounds like the issue you're describing.

[1]: https://github.com/cilindrox/hapi-ioredis/blob/master/lib/index.js#L44

**Comment by cilindrox** [comment]
Closed via #9
